velox/exec/tests/RowContainerTest.cpp

```diff
@@ -32,44 +32,149 @@ using namespace facebook::velox::test;
 
 class RowContainerTest : public exec::test::RowContainerTestBase {
  protected:
-  void testExtractColumnForOddRows(
+  void testExtractColumn(
       RowContainer& container,
       const std::vector<char*>& rows,
       int column,
-      const VectorPtr& expected) {
+      const VectorPtr& expected,
+      int offset = 0) {
+    // This set of tests uses all variations of the extractColumn API to copy
+    // column value for all rows in the container.
+    testExtractColumnAllRows(container, rows, column, expected, offset);
+    // This set of tests use extractColumn with an out-of-order pattern
+    // and repetitions.
+    testExtractColumnOutOfOrder(container, rows, column, expected, offset);
+  }
+
+  void testExtractColumnAllRows(
+      RowContainer& container,
+      const std::vector<char*>& rows,
+      int column,
+      const VectorPtr& expected,
+      int offset = 0) {
     auto size = rows.size();
 
-    // Extract only odd rows.
-    std::vector<char*> oddRows(size, nullptr);
-    for (auto i = 1; i < size; i += 2) {
-      oddRows[i] = rows[i];
-    }
+    auto testBasic = [&](vector_size_t offset = 0) -> void {
+      // Test extractColumn from offset 0.
+      auto result = BaseVector::create(expected->type(), size, pool_.get());
+      container.extractColumn(rows.data(), size, column, result);
+      assertEqualVectors(expected, result);
 
-    auto result = BaseVector::create(expected->type(), size, pool_.get());
-    container.extractColumn(oddRows.data(), size, column, result);
-    EXPECT_EQ(size, result->size());
-    for (vector_size_t row = 0; row < size; ++row) {
-      if (row % 2 == 0) {
-        EXPECT_TRUE(result->isNullAt(row)) << "at " << row;
-      } else {
-        EXPECT_TRUE(expected->equalValueAt(result.get(), row, row))
-            << "at " << row << ": expected " << expected->toString(row)
-            << ", got " << result->toString();
+      // Test extractColumn from offset.
+      container.extractColumn(
+          rows.data(), size - offset, column, offset, result);
+      EXPECT_EQ(result->compare(expected.get(), offset, 0), 0);
```

Comment:
Shouldn't we check all rows ?

Reply:
We did not copy from positions 0 to offset. So not comparing those values.

Reply:
I believe "compare" compares only one row. Did we extract only one row? I through we extracted multiple rows, hence, expected we would compare all extracted rows. Am I missing something?

Reply:
My bad. I misunderstood the compare function. Have changed this now.